### ISSUE TITLE
Editor freezes when creating a new project after moving Godot to the second screen

### ISSUE THREAD

**Issue by nightblade9** [godotengine/godot#87238]
### Tested versions

4.2.1 stable official

### System information

Windows 10, NVidia GeForce GTX 1650 Ti, apparently OpenGL

### Issue description

When creating a new project, the editor window freezes:

![image](https://github.com/godotengine/godot/assets/31222662/3a556df0-fdee-4fc2-9806-c9c83b71d50b)

There is nothing of note in the console log, from what I can tell; the first two warnings are because I'm using the NuGet version, where the EXE filename isn't case-matching.

```
WARNING: Case mismatch opening requested file 'godot_v4.2.1-stable_win64.exe', stored as 'Godot_v4.2.1-stable_win64.exe.
     at: open_internal (drivers/windows/file_access_windows.cpp:132)
WARNING: Case mismatch opening requested file 'godot_v4.2.1-stable_win64.exe', stored as 'Godot_v4.2.1-stable_win64.exe.
     at: open_internal (drivers/windows/file_access_windows.cpp:132)
Godot Engine v4.2.1.stable.official.b09f793f5 - https://godotengine.org
OpenGL API 3.3.0 NVIDIA 532.09 - Compatibility - Using Device: NVIDIA - NVIDIA GeForce GTX 1650 Ti

Project is missing: c:/temp/audiotest/project.godot
```

### Steps to reproduce

- Set up your laptop with an external monitor, and Windows set to "Extend display"
- Open Godot
- Move Godot to the monitor screen
- Click the new project button
- Wait. Forever.

Note that creating a project on the first/default (laptop/internal) screen works fine.

### Minimal reproduction project (MRP)

Not applicable.

**Comment by Calinou** [comment]
- Related to https://github.com/godotengine/godot/issues/74261.

Can you reproduce this if you force Godot to start on integrated graphics? This is likely a NVIDIA Optimus issue.

**Comment by nightblade9** [comment]
> Can you reproduce this if you force Godot to start on integrated graphics? This is likely a NVIDIA Optimus issue.

Yes. I followed [these instructions](https://www.itechtics.com/use-specific-gpu/), and added every instance of Godot to use the integrated graphics driver. (The actual Godot .exe, which is 100MB+, as well as the Chocolatey wrapper EXE, and the godot console EXE).

**Comment by Ruruie-FGSC** [comment]
This is replicable on a desktop 4090 with the latest NVIDIA 546.33 studio drivers and is not limited to a New Project. It will trigger on any action that tries to generate a pop-up element after moving the Project Manager window onto a second screen and even when both monitors are set to the same resolution and orientation.

**Works**
- Mousing over the Asset Library Projects tab or any Project or Button (causing them to partial highlight).
- Selecting a Project (causing a full highlight).
- Switching between Projects and Asset Library Projects tabs.

**Crashes**
- New, Import, Scan, or About (causes a pop-up window).
- [en] English, or Sort (causes a dropdown style pop-up).

Once you open a window on your main monitor, you can drag it over to the new monitor without issue. Any of the above crash inducing functionalities (such as the Version Control Metadata dropdown menu within New Project) will work on your main monitor but crash from the second. The same principles will still apply when you load up a Project. One thing that causes a crash there is mousing over any resource, which causes a tooltip to pop up, which will crash Godot if it's in the second monitor. Enabling Single Window Mode or, conversely, disabling Multi Window Mode will fully protect Godot; you can move the Project window onto either monitor and everything works fine in either. However, the Project Manager window can't be protected in this way and will always crash.

**Comment by Zireael07** [comment]
Some people have had a similar freeze on some mobile devices that might be related

**Comment by nightblade9** [comment]
@Ruruie-FGSC thanks for the detailed description of what works and what doesn't!

The other work-around is simply not to do anything with Godot until you get your project loaded.

**Comment by Ruruie-FGSC** [comment]
Recently discovered this could be renderer dependent. 

The editor does not crash on Forward+ but will on Compatibility. 

Project List seems to always be in Compatibility renderer however.
